apis/python/src/tiledbsoma/utils.py

```diff
@@ -0,0 +1,22 @@
+import numpy as np
+import pandas as pd
+from tiledbsoma import pytiledbsoma as clib
+from typing import Any
+import tiledb
+from .options import SOMATileDBContext
+from .options._soma_tiledb_context import _validate_soma_tiledb_context
+
+
+def build_index(keys: np.typing.NDArray[np.int64], context: SOMATileDBContext) -> clib.IntIndexer:
+    if len(np.unique(keys)) != len(keys):
+        raise pd.errors.InvalidIndexError(
+            "Reindexing only valid with uniquely valued Index objects"
+        )
+    # If we chose to keep `context` as optional, we would add this line here:
+    #     context = context or options.SOMATileDBContext()
+    tdb_concurrency = int(context.tiledb_ctx.config().get("sm.compute_concurrency_level", _validate_soma_tiledb_context(
```

Comment:
The "default" parameter to `.get` should just be the default value we want (or, in this case, twice the number of threads we want since we divide it by two below). If there's no `sm.compute_concurrency_level` entry in the config, this is going to try to convert a SOMATileDBContext into an int, which will fail.